src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -244,21 +246,31 @@ private async Task CollectActivationsImpl(bool scanStale, TimeSpan ageLimit = de
             watch.Start();
             var number = Interlocked.Increment(ref collectionNumber);
             long memBefore = GC.GetTotalMemory(false) / (1024 * 1024);
-            logger.Info(ErrorCode.Catalog_BeforeCollection, "Before collection#{0}: memory={1}MB, #activations={2}, collector={3}.",
-                number, memBefore, activations.Count, this.activationCollector.ToString());
-            List<ActivationData> list = scanStale ? this.activationCollector.ScanStale() : this.activationCollector.ScanAll(ageLimit);
-            collectionCounter.Increment();
-            var count = 0;
-            if (list != null && list.Count > 0)
+
+            if (memBefore < memoryThresholdInMegabytes)
+            {
+                watch.Stop();
+                logger.Info(ErrorCode.Catalog_CollectionSuppressed,
+                    $"Collection suppressed#{number}: memory={memBefore}MB, #activations={activations.Count}, memoryPressureThreshold={memoryThresholdInMegabytes}, suppression time={watch.Elapsed}.");
+            }
+            else
             {
-                count = list.Count;
-                if (logger.IsEnabled(LogLevel.Debug)) logger.Debug("CollectActivations{0}", list.ToStrings(d => d.Grain.ToString() + d.ActivationId));
-                await DeactivateActivationsFromCollector(list);
+                logger.Info(ErrorCode.Catalog_BeforeCollection, "Before collection#{0}: memory={1}MB, #activations={2}, collector={3}.",
```

Comment:
Be careful, the new logging system used in 2.0 and forward does not support index based string for logging (yes we are still using this in some part of orleans…)

More infos: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.1#log-message-template